src/__template__/ComponentTemplate.vue

```diff
@@ -90,6 +101,70 @@ const rootElementStyles = computed((): Record<string, string> => ({
   lineHeight: '0',
   width: iconSize.value,
 }))
+
+/**
+ * Prefix all ids within the SVG element and update all corresponding references to these ids. This is useful to avoid id conflicts when utilizing multiple instances of the same SVG in the DOM.
+ *
+ * @param {SVGElement} svgElement - The SVG element whose potential IDs need to be prefixed.
+ */
+const prefixSvgIds = (svgElement: SVGElement): void => {
+  if (!svgElement) {
+    return
+  }
+
+  const defsElement = svgElement.querySelector('defs')
+
+  // Prepare a map to store the original and new IDs for quick lookups
+  const idMap: Record<string, string> = {}
+
+  if (defsElement) {
+    // Prefix all direct children of <defs> that have an `id`
+    defsElement.querySelectorAll('[id]').forEach((element) => {
+      const originalId = element.getAttribute('id')
+      const newId = `${Math.random().toString(36).substring(2, 12)}-${originalId}`
+
+      // Map old id to new id
+      idMap[originalId!] = newId
+
+      // Update the element's id with the new prefixed id
+      element.setAttribute('id', newId)
+    })
+  }
+
+  // Update all references to these prefixed ids in attributes like `url(#id)`, `href`, `xlink:href`
+  const referencingAttributes = ['fill', 'stroke', 'filter', 'mask', 'clip-path', 'xlink:href', 'href']
+
+  // Function to update references to ids in attributes
+  const updateReferences = (element: Element): void => {
+    referencingAttributes.forEach((attr) => {
+      const attrValue = element.getAttribute(attr)
+      if (attrValue) {
+        // Match any url(#id) or href="#id" references
+        const updatedValue = attrValue.replace(/url\(#([^)]+)\)/g, (match, id) => {
+          return idMap[id] ? `url(#${idMap[id]})` : match
+        }).replace(/#([^\s]+)/g, (match, id) => {
+          return idMap[id] ? `#${idMap[id]}` : match
+        })
+
+        // If the value changed, update the attribute
+        if (updatedValue !== attrValue) {
+          element.setAttribute(attr, updatedValue)
+        }
+      }
+    })
+  }
+
+  // Traverse all elements in the SVG to update id references
+  svgElement.querySelectorAll('*').forEach((element) => {
+    updateReferences(element)
+  })
+}
+
+onMounted(() => {
+  if (props.randomizeIds && svgElement.value) {
+    prefixSvgIds(svgElement.value)
+  }
+})
```

Comment:
There is still a problem with SSR before hydration is done, which I believe can't be resolved if we continue with this approach.

Reply:
I may be able to do processing in the setup block via a render function to solve for SSR. I'll try that this morning

Reply:
Updated for SSR if you want to take another look